### ISSUE TITLE
Implement parameter optimization for IndependentBayesianScorer

### ISSUE THREAD

**Issue by athola** [athola/importobot#64]
### 📋 Summary

The `EvidenceAccumulator.optimize_parameters()` method currently logs a message that "Parameter optimization not yet implemented for IndependentBayesianScorer". This feature would enable the system to learn optimal Bayesian parameters from training data, improving format detection accuracy over time.

### 📍 Current Location

**File**: `src/importobot/medallion/bronze/evidence_accumulator.py:378-399`

**Current Implementation**:
```python
def optimize_parameters(self, training_data: List[Tuple[str, float]]) -> None:
    """Optimize weighted evidence parameters using training data.

    Args:
        training_data: List of (format_name, expected_confidence) pairs
    """
    # Convert training data to EvidenceMetrics format
    scorer_training_data = []

    for format_name, expected_confidence in training_data:
        if format_name in self.evidence_profiles:
            profile = self.evidence_profiles[format_name]
            metrics = self._profile_to_metrics(profile)
            scorer_training_data.append((metrics, expected_confidence))

    if scorer_training_data:
        # Note: Parameter optimization not yet implemented
        # for IndependentBayesianScorer
        logger.info(
            "Parameter optimization requested but not yet implemented "
            "for IndependentBayesianScorer"
        )
```

### 🎯 Acceptance Criteria

1. **Parameter Optimization Method**: Implement `optimize_parameters()` method in `IndependentBayesianScorer`
2. **Optimization Algorithm**: Use Maximum Likelihood Estimation (MLE) or Bayesian optimization
3. **Configurable Parameters**: Optimize key parameters in `IndependentBayesianParameters`:
   - Beta distribution parameters for quality, completeness, uniqueness
   - Evidence amplification factors
   - Numerical stability bounds
4. **Training Data Format**: Support training data as `(EvidenceMetrics, expected_confidence)` pairs
5. **Validation**: Include parameter validation and convergence checking
6. **Persistence**: Allow saving/loading optimized parameters
7. **Performance**: Efficient optimization that can handle reasonable training datasets
8. **Testing**: Comprehensive unit tests for the optimization functionality

#### Parameters to Optimize

From `IndependentBayesianParameters` class:
```python
# Beta distribution parameters for evidence quality
quality_alpha: float = 3.0
quality_beta: float = 1.5

# Beta distribution parameters for completeness  
completeness_alpha: float = 4.0
completeness_beta: float = 1.0

# Beta distribution parameters for uniqueness
uniqueness_alpha: float = 3.0
uniqueness_beta: float = 1.5

# Numerical stability parameters
min_log_likelihood: float = -20.0
max_log_likelihood: float = 2.0
```

From `BayesianConfiguration` class:
```python
# P(E|¬H) Estimation Parameters
min_evidence_not_format: float = 0.01
evidence_not_format_scale: float = 0.49
evidence_not_format_exponent: float = 2.0
numerical_epsilon: float = 1e-15
```

#### Suggested Implementation Approach

1. **Optimization Method**: Use `scipy.optimize.minimize` with appropriate objective function
2. **Objective Function**: Minimize negative log-likelihood or Mean Squared Error between predicted and expected confidences
3. **Parameter Bounds**: Define reasonable bounds for each parameter to maintain mathematical validity
4. **Gradient Information**: Use gradient-based optimization for faster convergence
5. **Validation**: Include cross-validation to prevent overfitting

#### Integration Points

1. **Update `IndependentBayesianScorer`**: Add `optimize_parameters()` method
2. **Update `EvidenceAccumulator`**: Connect to the scorer's optimization method
3. **Configuration**: Add optimization parameters to configuration
4. **Persistence**: Add methods to save/load optimized parameters

### 🧪 Testing Requirements

1. **Unit Tests**: Test optimization convergence and parameter updates
2. **Integration Tests**: Test end-to-end parameter optimization workflow
3. **Performance Tests**: Ensure optimization completes in reasonable time
4. **Validation Tests**: Test parameter bounds and mathematical constraints
5. **Regression Tests**: Ensure optimization doesn't break existing functionality

### 📊 Success Metrics

- Parameter optimization converges within reasonable iterations
- Optimized parameters improve detection accuracy on validation data
- Optimization completes within acceptable time limits (e.g., < 30 seconds for typical datasets)
- Mathematical constraints are maintained (probabilities stay in [0,1], etc.)

### 🏷️ Labels

enhancement, machine-learning, bayesian-optimization, format-detection, parameter-tuning

### 🔗 Related Issues

# EvidenceAccumulator interface already prepared for parameter optimization
# IndependentBayesianScorer has mathematical foundation ready for optimization
# Beta distribution parameters are well-suited for MLE optimization

### 📚 References

- **Maximum Likelihood Estimation for Beta Distributions**: Mathematical foundation for optimizing Beta parameters
- **Bayesian Optimization**: For parameter space exploration
- **Scipy Optimization**: `scipy.optimize.minimize` for numerical optimization
- **Cross-Validation**: To prevent overfitting during optimization

### 🚀 Implementation Priority

**High**: This feature would significantly improve the system's ability to adapt to specific datasets and use cases, making format detection more accurate over time.

### 💡 Additional Notes

- The current Bayesian framework is well-designed for parameter optimization
- Beta distribution parameters have clear mathematical properties for optimization
- Evidence metrics provide good features for learning optimal parameters
- The system already has the infrastructure to collect training data
- Consider adding logging and monitoring for optimization progress
